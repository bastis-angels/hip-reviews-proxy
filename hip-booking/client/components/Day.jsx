import React from 'react'


const Square = styled.div`
  &:hover {
    color: #fff;
    background-color: #40d9ac;
  }
  ${props => props.selected && 'color: #fff; background-color: #40d9ac;'}
  font-size: 1.2em;
  font-weight: 300;
  position: relative;
  color: #333333;
  border: 1px solid #ebebeb;
  width: 36px;
  height: 36px;
  border-radius: 0;
  text-align: center;
  text-shadow: none;
  padding: 10px;
  /* ${ props => props.weekday && 'add ::after'} */
  /* ${ props => props.disabled && 'color: #757575; background-color: #ebebeb;' } */
`

export default function Day (props) {
  return (
    <Square 
      selected={typeof props.arrive !== 'string' && props.arrive.format() === props.moment.format()} 
      id={props.moment.format()} 
      onClick={props.handleDayClick}
    >
      {props.moment.date()}
    </Square>
  )
}