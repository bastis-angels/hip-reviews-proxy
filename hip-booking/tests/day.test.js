import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import Day from '../client/components/Day.jsx'
import moment from 'moment'


describe('Day', () => {
  it('displays text based on props.day', () => {
    const day = shallow(
      <Day 
        moment={moment('2019-03-01')}
        handleDayClick={() => null}
        arrive = {'Select date'}
      />);
    
    expect(day.text()).toBe('1');
    day.setProps({moment: moment('2019-04-02')});
    expect(day.text()).toBe('2');
  });

  it('calls props.handleDayClick on click event', () => {
    const handleClickSpy = jest.fn();

    const day = shallow(
      <Day 
        moment={moment()}
        handleDayClick={handleClickSpy}
        arrive = {'Select date'}
      />
    )
    day.simulate('click');
    expect(handleClickSpy).toHaveBeenCalled();
  });

  it('renders day based on state.moment', () => {
    const day = shallow(      
      <Day 
        moment={moment('2019-05-12')}
        handleDayClick={() => null}
        arrive = {'Select date'}
      />
    );
    expect(toJson(day)).toMatchSnapshot();
  });

});
