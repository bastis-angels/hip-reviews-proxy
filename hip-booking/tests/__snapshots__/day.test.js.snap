// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day renders day based on state.moment 1`] = `
<styled.div
  id="2019-05-12T00:00:00-07:00"
  onClick={[Function]}
  selected={false}
>
  12
</styled.div>
`;
